trio/_socket.py

```diff
@@ -472,44 +478,45 @@ def __getattr__(self, name):
             return getattr(self._sock, name)
         raise AttributeError(name)
 
-    def __dir__(self):
-        return super().__dir__() + list(self._forward)
+    def __dir__(self) -> Iterable[str]:
+        yield from super().__dir__()
+        yield from self._forward
 
-    def __enter__(self):
+    def __enter__(self) -> Self:
         return self
 
-    def __exit__(self, *exc_info):
+    def __exit__(self, *exc_info: object) -> None:
         return self._sock.__exit__(*exc_info)
 
     @property
-    def family(self):
+    def family(self) -> _stdlib_socket.AddressFamily:
         return self._sock.family
 
     @property
-    def type(self):
+    def type(self) -> _stdlib_socket.SocketKind:
         return self._sock.type
 
     @property
-    def proto(self):
+    def proto(self) -> int:
         return self._sock.proto
 
     @property
-    def did_shutdown_SHUT_WR(self):
+    def did_shutdown_SHUT_WR(self) -> bool:
         return self._did_shutdown_SHUT_WR
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return repr(self._sock).replace("socket.socket", "trio.socket.socket")
 
-    def dup(self):
+    def dup(self) -> _SocketType:
         """Same as :meth:`socket.socket.dup`."""
         return _SocketType(self._sock.dup())
 
-    def close(self):
+    def close(self) -> None:
         if self._sock.fileno() != -1:
             trio.lowlevel.notify_closing(self._sock)
             self._sock.close()
 
-    async def bind(self, address):
+    async def bind(self, address: tuple[Any, ...] | str | bytes) -> None:
```

Comment:
from `trio-typing`

Reply:
Mind trying `tuple[object, ...]` and seeing if that works?

Reply:
`str` certainly isn't a `tuple[object, ...]`, but it would match against `Iterable[object]`.

Reply:
Sorry I meant instead of tuple[Any, ...] :^)

Reply:
oh derp, yeah that works.